src/gluonts/evaluation/_base.py

```diff
@@ -458,18 +456,30 @@ def quantile_loss(target, quantile_forecast, q):
     def coverage(target, quantile_forecast):
         return np.mean((target < quantile_forecast))
 
+    @staticmethod
+    def vanilla_mase(target, forecast, seasonal_error):
+        r"""
+        .. math::
+
+            mase = mean(|Y - Y_hat|) / seasonal_error
+
+        https://www.m4.unic.ac.cy/wp-content/uploads/2018/03/M4-Competitors-Guide.pdf
+        """
+        return np.mean(np.abs(target - forecast)) / seasonal_error
+
     @staticmethod
     def mase(target, forecast, seasonal_error):
         r"""
+        Upper bound limited version of MASE, avoids infinite values.
+        For traditional implementation use vanilla_mase.
         .. math::
 
             mase = mean(|Y - Y_hat|) / seasonal_error
 
         https://www.m4.unic.ac.cy/wp-content/uploads/2018/03/M4-Competitors-Guide.pdf
         """
-        flag = seasonal_error <= Evaluator.zero_tol
-        return (np.mean(np.abs(target - forecast)) * (1 - flag)) / (
-            seasonal_error + flag
+        return np.mean(np.abs(target - forecast)) / max(
```

Comment:
In principle this is correct, but this would blow up the overall `MASE` if one time series in the data set has a very small seasonal error (e.g., constant time series). It does not reflect what we actually want because ideally such time series should be removed before evaluating the whole data set and we usually don’t that explicitly (e.g., `electricity` has one such time series).

The best thing here is to keep the current version of `MASE` as it as and introduce `MASE_strict` and have your version there. Keeping track of both these metrics during evaluations will tell us if the data set has any time series that is “bad” for `MASE` and choose either of them accordingly.

Reply:
> It does not reflect what we actually want because ideally such time series should be removed before evaluating the whole data set and we usually don’t that explicitly

That's a good point: those are the cases where your evaluation might be skewed by very simple time series, and also comparisons between models (and based on other metrics) should be looked into with suspicion

Reply:
Alrighty, thank you for the feedback @rshyamsundar. I'll implement this `MASE strict` version then. 
Would it make sense to implement similar alternative metrics for the other metrics which currently are implemented using the flag approach? (i.e. `MAPE_strict`, `sMAPE_strict` etc)